analyses/TX_cd_2020/03_sim_TX_cd_2020.R

```diff
@@ -0,0 +1,445 @@
+###############################################################################
+# Simulate plans for `TX_cd_2020`
+# © ALARM Project, February 2022
+###############################################################################
+set.seed(02138)
+library(sf)
+library(tidyverse)
+library(patchwork)
+
+source("analyses/TX_cd_2020/TX_helpers.R")
+source("analyses/TX_cd_2020/01_prep_TX_cd_2020.R")
+source("analyses/TX_cd_2020/02_setup_TX_cd_2020.R")
+
+cluster_pop_tol <- 0.0025
+
+map <- set_pop_tol(map, 0.01)
+nsims <- 5000
+
+diag_plots <- FALSE
+
+# Run the simulation -----
+cli_process_start("Running simulations for {.pkg TX_cd_2020}")
+
+# Unique ID for each row, will use later to reconnect pieces
+map$row_id <- 1:nrow(map)
+
+########################################################################
+# Cluster #1: Greater Houston
+
+clust1 <- c("Austin", "Brazoria", "Chambers", "Fort Bend",
+    "Galveston", "Harris", "Liberty", "Montgomery", "Waller")
+
+clust1 <- paste(clust1, "County")
+
+m1 <- map %>% filter(county %in% clust1)
+m1 <- set_pop_tol(m1, cluster_pop_tol)
+attr(m1, "pop_bounds") <-  attr(map, "pop_bounds")
+attr(m1, "pop_bounds")
+attr(map, "pop_bounds")
+attr(m1, "ndists")
+attr(map, "ndists")
+
+# Check for potential MMDs
+m1 %>% st_drop_geometry() %>%
+    filter(county %in% clust1) %>%
+    group_by(cd_2020) %>%
+    summarise(hisp_prop = sum(cvap_hisp)/sum(cvap),
+        black_prop = sum(cvap_black)/sum(cvap))
+
+map <- map %>%
+    mutate(cluster_edge = ifelse(row_id %in% m1$row_id, 1, 0))
+
+z <- geomander::seam_geom(map$adj, map, admin = "cluster_edge", seam = c(0, 1))
+
+z <- z[z$cluster_edge == 1, ]
+
+border_idxs <- which(m1$row_id %in% z$row_id)
+
+constraints <- redist_constr(m1) %>%
+    add_constr_grp_hinge(
+        20,
+        cvap_hisp,
+        total_pop = cvap,
+        tgts_group = c(0.45)
+    ) %>%
+    add_constr_grp_hinge(
+        10,
+        cvap_black,
+        total_pop = cvap,
+        tgts_group = c(0.45)) %>%
+    add_constr_custom(strength = 10, function(plan, distr) {
+        ifelse(any(plan[border_idxs] == 0), 0, 1)
+    })
+
+n_steps <- (sum(m1$pop)/attr(map, "pop_bounds")[2]) %>% floor()
+
+houston_plans <- redist_smc(m1, counties = county,
+    nsims = nsims, n_steps = n_steps,
+    constraints = constraints)
+
+#############################################################
+## Local Diagnostic plots
```

Comment:
let's wrap all of these diagnostic plots in an `if (interactive()) { }` or `if (FALSE)` block so that the script can run without generating them

Reply:
alternatively, there's so many, it may make sense to move to an diagnostics file that isn't run by default

Reply:
yes, I like that separate script thing!